src/main/clojure/clara/rules/engine.cljc

```diff
@@ -770,18 +767,13 @@
         nil)))
 
   (left-retract [node join-bindings tokens memory transport listener]
-    (doseq [:let [previous-results (mem/get-accum-reduced-all memory node join-bindings)
-                  ;; If there are any previous results we know we have an actual previous reduced
-                  ;; value because we have tokens to remove.  This means there were matches before
-                  ;; and a value is always accumulated and propagated when there are matches.
-                  ;; The converted value only needs to be calculated once for all token + element
-                  ;; pairs retracted.
-                  [_ [_ previous-reduced]] (first previous-results)
-                  previous-converted (when (some? previous-reduced)
-                                       ((:convert-return-fn accumulator) previous-reduced))
-                  removed-tokens (mem/remove-tokens! memory node join-bindings tokens)
+    (doseq [:let [removed-tokens (mem/remove-tokens! memory node join-bindings tokens)
                   remaining-tokens (mem/get-tokens memory node join-bindings)
 
+                  ;; Note:  Memory *must* be read here before the memory is potentially cleared in the
+                  ;; following lines.
```

Comment:
I actually broke things at one point by flipping the two lines.  So I put the comment to avoid that happening again.  :P